crates/walrus-service/src/client/cli/args.rs

```diff
@@ -1113,6 +1123,31 @@ pub struct FileOrBlobId {
     pub(crate) blob_id: Option<BlobId>,
 }
 
+/// CLI enum for selecting upload presets. Converted to SDK UploadMode at runtime.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize, clap::ValueEnum)]
+#[serde(rename_all = "kebab-case")]
+pub enum UploadModeCli {
+    Conservative,
+    Balanced,
+    Aggressive,
+}
+
+impl Default for UploadModeCli {
+    fn default() -> Self {
+        Self::Balanced
+    }
+}
+
+impl From<UploadModeCli> for UploadMode {
+    fn from(value: UploadModeCli) -> Self {
+        match value {
+            UploadModeCli::Conservative => UploadMode::Conservative,
+            UploadModeCli::Balanced => UploadMode::Balanced,
+            UploadModeCli::Aggressive => UploadMode::Aggressive,
+        }
+    }
+}
```

Comment:
Question: Why do we need a separate struct here? Couldn't we use `UploadMode` directly?

Reply:
Using UploadMode directly from the CLI would drag a clap dependency into the SDK, so I kept a thin UploadModeCli wrapper and convert to UploadMode at runtime.

Reply:
That makes sense. It still feels a bit suboptimal to duplicate the code, but I didn't find a cleaner way... 😢